### PR TITLE
Create package skeleton

### PR DESCRIPTION
**Pull request by squahtx**
Created by using https://github.com/matrix-org/synapse-module-cookiecutter-template followed by some manual edits.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+python_requires = >= 3.7
+install_requires =
+  attrs
+
+
+[options.extras_require]
+dev =
+  # for tests
+  tox
+  twisted
```
Comment (squahtx) [inline setup.cfg:17]: Do we want a dependency on `twisted` for this? It feels a bit heavyweight for a utility library.

I suppose it's a dev dependency so it's alright.

Comment (DMRobertson) [reply setup.cfg:17]: There's code shared between Synapse and Sydent that I'd hoped to pull into here (#2, #3) which extends or interfaces with twisted. Maybe they'd be better off in a separate library?

Comment (squahtx) [reply setup.cfg:17]: Can we maybe do some magic to make twisted an optional dependency?

Comment (DMRobertson) [reply setup.cfg:17]: extra_requires I think? https://setuptools.pypa.io/en/latest/userguide/declarative_config.html

Comment (callahad) [reply setup.cfg:17]: Extras are the way to go, here. There's a really excellent blog post discussing how someone used this approach (with some creative nuance) for a scientific computing utility library... let me see if I can dig that up (don't block on this beyond Monday; I'll either find it tonight or I won't be able to find it at all)

Comment (callahad) [reply setup.cfg:17]: Not finding it. If I recall correctly, it did something simple but clever around checking whether optional libraries were available before exposing functions / submodules which relied on them... Oh well.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      - run: tox -e check_types
+
+  unit-tests:
+    name: Unit tests
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # Run the unit tests both against our oldest supported Python version
+        # and the newest stable.
+        python_version: [ "3.7", "3.x" ]
```
Comment (DMRobertson) [inline .github/workflows/ci.yml:37]: Synapse's oldest supported version is currently 3.6... though not for long.

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+
```
Comment (DMRobertson) [inline .gitlab-ci.yml]: Don't know if we want this? /shrug

File: setup.cfg (hunk 1 above)
Comment (DMRobertson) [reply setup.cfg:17]: Let's leave it as it is. It's called "Common utilities for Synapse, Sydent and Sygnal" and all three of those projects use twisted. (IMO)

File: mypy.ini [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[mypy]
+strict = true
```
Comment (DMRobertson) [inline mypy.ini:2]: phwoar!

**Review by DMRobertson — APPROVED** [review]
LGTM; a few thoughts, but nothing blocking it.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (squahtx) [reply .github/workflows/ci.yml:37]: EOL for 3.6 is in 3 weeks so I'm inclined to leave it as it is

File: .gitlab-ci.yml (hunk 3 above)
Comment (squahtx) [reply .gitlab-ci.yml]: Good point, let's remove it.

**Review by DMRobertson — APPROVED** [review]

**Pull request merged**